### PR TITLE
Make library slugs optional during Lib Creation

### PR DESCRIPTION
**Pull request by ryanpbrewster**
If a library slug is not provided, we will generate a valid one on the backend (based on the name, with a random suffix to ensure no collisions).

@andrewconner @camhashemi @leogrim

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackTeamCommander.scala [hunk 1]
```diff
@@ -140,7 +140,6 @@ class SlackTeamCommanderImpl @Inject() (
         case UserSpace(_) => LibraryVisibility.SECRET
         case OrganizationSpace(_) => LibraryVisibility.ORGANIZATION
       },
-      slug = LibrarySlug.generateFromName(libraryName),
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackTeamCommander.scala:143]: This was the thing that prompted this change.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala [hunk 2]
```diff
@@ -165,7 +161,15 @@ class LibraryCommanderImpl @Inject() (
       case UserSpace(_) => None
       case OrganizationSpace(orgId) => Some(orgId)
     }
-    val newSlug = LibrarySlug(libCreateReq.slug)
+    val newSlug = libCreateReq.slug match {
+      case Some(slug) => LibrarySlug(slug)
+      case None =>
+        val suffixes = "" +: Stream.continually("-" + RandomStringUtils.randomNumeric(2)).take(9)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala:167]: Any chance we could only do a suffix if it's needed?

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala:167]: That's what the `""` does.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala:167]: Ah, carry on :)

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala [hunk 3]
```diff
@@ -165,7 +161,15 @@ class LibraryCommanderImpl @Inject() (
...
+        val baseSlug = LibrarySlug.generateFromName(libCreateReq.name)
+        suffixes.map(suff => LibrarySlug(baseSlug + suff)).filter { slug =>
+          libraryRepo.getBySpaceAndSlug(targetSpace, slug).isEmpty
+        }.head
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala:171]: this code is like a Stream come true

**Pull request merged**